wb/nm_helper/connection_manager.py

```diff
@@ -315,45 +317,54 @@ def cycle_loop(self):
             self.deactivate_lesser_gsm_connections(new_connection, new_tier)
             self.apply_metrics()
 
+    def check_current_connection(self):
+        logging.debug("checking currently active connection %s", self.current_connection)
+        try:
+            active_cn = self.find_activated_connection(self.current_connection)
+            if active_cn and check_connectivity(active_cn, self.config):
+                logging.debug(
+                    "Current connection %s is most preferred and has connectivity",
+                    self.current_connection,
+                )
+                return True
+        except dbus.exceptions.DBusException as ex:
+            self._log_connection_check_error(self.current_connection, ex)
+        return False
+
+    def check_non_current_connection(self, tier, cn_id):
+        if (
+            self.current_tier
+            and tier.priority == self.current_tier.priority
+            and cn_id == self.current_connection
+        ):
+            logging.debug("current connection %s was already checked before, skipping", cn_id)
+            return False
+        logging.debug("checking connection %s", cn_id)
+        try:
+            active_cn = self.find_activated_connection(cn_id)
+            if not active_cn and self.ok_to_activate_connection(cn_id):
+                active_cn = self.activate_connection(cn_id)
+                self.timeouts.touch_connection_retry_timeout(cn_id)
+            if active_cn and check_connectivity(active_cn, self.config):
+                return True
+        except dbus.exceptions.DBusException as ex:
+            self._log_connection_check_error(cn_id, ex)
+            self.timeouts.touch_connection_retry_timeout(cn_id)
+        return False
+
     def check(self) -> (ConnectionTier, str, bool):
         logging.debug("check(): starting iteration")
         self.timeouts.debug_log_timeouts()
         for tier in self.config.tiers:
             logging.debug("checking tier %s", tier.name)
             # first, if tier is current, check current connection
             if self.current_tier and self.current_connection and self.current_tier.priority == tier.priority:
-                logging.debug("checking currently active connection %s", self.current_connection)
-                try:
-                    active_cn = self.find_activated_connection(self.current_connection)
-                    if active_cn and check_connectivity(active_cn, self.config):
-                        logging.debug(
-                            "Current connection %s is most preferred and has connectivity",
-                            self.current_connection,
-                        )
-                        return self.current_tier, self.current_connection
-                except dbus.exceptions.DBusException as ex:
-                    self._log_connection_check_error(self.current_connection, ex)
+                if self.check_current_connection():
```

Comment:
Читается не очень, на мой взгляд было бы лучше так:
```suggestion
                if self.current_connection_is_ok():
```
А лучше что-то более содержательное: что именно тут требуется от соединения, чтобы оно прошло проверку.

В общем, чтобы функция `check` читалась как текст

Reply:
сделал красивее